proxy.go

```diff
@@ -643,26 +643,63 @@ func (rp *reverseProxy) getUser(name string, password string) (found bool, u *us
 	defer rp.lock.RUnlock()
 	found = false
 	u = rp.users[name]
-	if u != nil {
+	switch {
+	case u != nil:
 		found = (u.password == password)
 		// existence of c and cu for toCluster is guaranteed by applyConfig
 		c = rp.clusters[u.toCluster]
 		cu = c.users[u.toUser]
-	} else if rp.hasWildcarded {
-		if s := strings.Split(name, "_"); len(s) == 2 {
-			u = rp.users[s[0]+"_*"]
-			if u != nil && u.isWildcarded {
-				c = rp.clusters[u.toCluster]
-				wildcardedCu := c.users[u.toUser]
-				if wildcardedCu != nil {
-					found = true
-					cu = wildcardedCu
-					cu.name = name
-					cu.password = password
+	case name == "" || name == defaultUser:
+		// default user can't work with the wildcarded feature for security reasons
```

Comment:
Can you elaborate in the comment what are the security scenarios we're talking about? I'm going to forget it soon as I know myself 😄

Reply:
if the clickhouse server is ill-configured and the default user is usable it's not a big deal currently since "external" users have to use chproxy which doesn't allow the default user unless the chproxy admin created a default user with no password.
But, with the wildcarded feature and more specifically the `*` case, an external user/hacker could try to connect to clickhouse using the usr/pwd "default"/"" and access data is not supposed to access.

Reply:
Could you add that to the doc or comment?